src/common/Command.cpp

```
@@ -377,7 +377,12 @@ namespace Cmd {
     }
 
     const std::string& Args::Argv(int argNum) const {
-        return args[argNum];
+        static std::string empty;
```

I don't think we should mask programming errors like this. I think out of bounds accesses should crash. There is no reason that a programmer should attempt to access an out of bounds item.

---

Well there is no guarantee it crashes, you can get garbage data instead which is a lot less nice. I don't want to have the undefined behavior here since it would be easy to miss when porting an old-style command. I added an assert so it will crash in debug mode.

---

I think @DolceTriade means that the ASSERT should be enough.

---

ya

---

You want it to silently return garbage in release mode?

---

Yes